__tests__/app.test.js

```diff
@@ -53,18 +53,53 @@ describe("GET -  /api/articles/:article_id", () => {
       .expect(200)
       .then(({ body }) => {
         const { article } = body;
-        console.log(article)
-    
+
         expect(article).toHaveProperty("title", expect.any(String));
         expect(article).toHaveProperty("article_id", expect.any(Number));
         expect(article).toHaveProperty("body", expect.any(String));
         expect(article).toHaveProperty("topic", expect.any(String));
         expect(article).toHaveProperty("created_at", expect.any(String));
         expect(article).toHaveProperty("votes", expect.any(Number));
-        expect(article).toHaveProperty(
-          "article_img_url",
-          expect.any(String)
-        );
+        expect(article).toHaveProperty("article_img_url", expect.any(String));
+      });
+  });
+});
+test("400 : Responds with message -'Bad Request' for an invalid article id", () => {
+  return request(app)
+    .get("/api/articles/notAnId")
+    .expect(400)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Bad Request");
+    });
+});
+test("404 : Responds with message -'Not Found' when article id is valid but does not exist", () => {
+  return request(app)
+    .get("/api/articles/99999999")
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toBe("Not Found");
+    });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: Responds with new comment after adding it to the database", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "Impressive article",
+    };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        console.log(comment.body);
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("body", "Impressive article");
+        expect(comment).toHaveProperty("article_id", 5);
+        expect(comment).toHaveProperty("author", "butter_bridge");
+        expect(comment).toHaveProperty("votes", 0);
+        expect(comment).toHaveProperty("created_at", expect.any(String));
       });
   });
```

Comment:
There are alot of tests for this endpoint I'm afraid 😭

You are missing two: 

404 - Username not found (the username must exist in the database, psql will throw it's own error for you to catch) 

400 - Missing keys/malformed request - For example const newComment = {
      notusername: "butter_bridge",
      notbody: "Impressive article",
    }; or const newComment = {}; 
    
    again psql will throw it's own error here I think